packages/tools/client-debugger/client-debugger-view/src/DevtoolsView.tsx

```diff
@@ -0,0 +1,449 @@
+ * and sub-views.
+ */
+export function DevtoolsView(): React.ReactElement {
+	// Set of features supported by the Devtools.
+	const [supportedFeatures, setSupportedFeatures] = React.useState<
+		DevtoolsFeatureFlags | undefined
+	>();
+
+	const messageRelay = useMessageRelay();
+
+	React.useEffect(() => {
+		/**
+		 * Handlers for inbound messages related to the registry.
+		 */
+		const inboundMessageHandlers: InboundHandlers = {
+			[DevtoolsFeatures.MessageType]: (untypedMessage) => {
+				const message = untypedMessage as DevtoolsFeatures.Message;
+				setSupportedFeatures(message.data.features);
+				return true;
+			},
+		};
+
+		/**
+		 * Event handler for messages coming from the Message Relay
+		 */
+		function messageHandler(message: Partial<ISourcedDevtoolsMessage>): void {
+			handleIncomingMessage(message, inboundMessageHandlers, {
+				context: loggingContext,
+			});
+		}
+
+		messageRelay.on("message", messageHandler);
+
+		// Query for supported feature set
+		messageRelay.postMessage(getSupportedFeaturesMessage);
+
+		return (): void => {
+			messageRelay.off("message", messageHandler);
+		};
+	}, [messageRelay, setSupportedFeatures]);
+
+	return (
+		<FluentProvider theme={getFluentUIThemeToUse()}>
+			{supportedFeatures === undefined ? (
+				<Waiting />
+			) : (
+				<_DevtoolsView supportedFeatures={supportedFeatures} />
+			)}
+		</FluentProvider>
+	);
+}
+
+interface _DevtoolsViewProps {
+	/**
+	 * Set of features supported by the Devtools.
+	 */
+	supportedFeatures: DevtoolsFeatureFlags;
+}
+
+/**
+ * Internal {@link DevtoolsView}, displayed once the supported feature set has been acquired from the webpage.
+ */
+function _DevtoolsView(props: _DevtoolsViewProps): React.ReactElement {
+	const { supportedFeatures } = props;
+
+	const [containers, setContainers] = React.useState<ContainerMetadata[] | undefined>();
+	const [menuSelection, setMenuSelection] = React.useState<MenuSelection | undefined>();
+
+	const messageRelay = useMessageRelay();
+
+	React.useEffect(() => {
+		/**
+		 * Handlers for inbound messages related to the registry.
+		 */
+		const inboundMessageHandlers: InboundHandlers = {
+			[ContainerList.MessageType]: (untypedMessage) => {
+				const message = untypedMessage as ContainerList.Message;
+				setContainers(message.data.containers);
+				return true;
+			},
+		};
+
+		/**
+		 * Event handler for messages coming from the Message Relay
+		 */
+		function messageHandler(message: Partial<ISourcedDevtoolsMessage>): void {
+			handleIncomingMessage(message, inboundMessageHandlers, {
+				context: loggingContext,
+			});
+		}
+
+		messageRelay.on("message", messageHandler);
+
+		// Query for list of Containers
+		messageRelay.postMessage(getContainerListMessage);
+
+		return (): void => {
+			messageRelay.off("message", messageHandler);
+		};
+	}, [messageRelay, setContainers]);
+
+	return (
+		<FluentProvider theme={getFluentUIThemeToUse()}>
```

Comment:
Unnecessary here? The parent (DevToolsView) already wraps this one with the theme provider.

Reply:
Copy-pasta strikes again! Will clean up :)